src/iop/lens.c

```diff
@@ -874,37 +872,56 @@ void modify_roi_in(struct dt_iop_module_t *self, struct dt_dev_pixelpipe_iop_t *
 
   if(modflags & (LF_MODIFY_TCA | LF_MODIFY_DISTORTION | LF_MODIFY_GEOMETRY | LF_MODIFY_SCALE))
   {
-    // acquire temp memory for distorted pixel coords
-    const size_t bufsize = (size_t)roi_in->width * 2 * 3;
-    void *const buf = dt_alloc_align(16, bufsize * dt_get_num_threads() * sizeof(float));
+    const int xoff = roi_in->x;
+    const int yoff = roi_in->y;
+    const int width = roi_in->width;
+    const int height = roi_in->height;
+    const int awidth = abs(width);
+    const int aheight = abs(height);
+    const int xstep = (width < 0) ? -1 : 1;
+    const int ystep = (height < 0) ? -1 : 1;
+
+    const size_t nbpoints = 2 * awidth + 2 * aheight;
+
+    float *buf = dt_alloc_align(16, nbpoints * 2 * 3 * sizeof(float));
```

Comment:
Show what? That it compiles? Sure: (yes, this is with OpenMP enabled)
```
~/darktable/build$ make lens
[  0%] Built target validate-cameras.xml
[ 31%] Built target rawspeed
[ 31%] Built target lautoc
[ 31%] Updating version string (git checkout)
Version string: 2.3.0+204~g7ba90c95b-dirty
[ 31%] Built target create_version_gen
[ 31%] Built target generate_version
[ 31%] Built target generate_metadata
[ 31%] Checking validity of data/darktableconfig.xml
[ 31%] Built target validate_darktableconfig_xml
[ 31%] Built target generate_preferences
[ 31%] Built target rawspeed_static
[ 95%] Built target lib_darktable
[100%] Built target lens
~/darktable/build$ git --no-pager diff -p -U8
```
```diff
$ git --no-pager diff -p -U8
diff --git a/src/iop/lens.c b/src/iop/lens.c
index 26c15d853..0350ffa25 100644
--- a/src/iop/lens.c
+++ b/src/iop/lens.c
@@ -878,20 +878,20 @@ void modify_roi_in(struct dt_iop_module_t *self, struct dt_dev_pixelpipe_iop_t *
     const int height = roi_in->height;
     const int awidth = abs(width);
     const int aheight = abs(height);
     const int xstep = (width < 0) ? -1 : 1;
     const int ystep = (height < 0) ? -1 : 1;
 
     const size_t nbpoints = 2 * awidth + 2 * aheight;
 
-    float *buf = dt_alloc_align(16, nbpoints * 2 * 3 * sizeof(float));
+    float *const buf = dt_alloc_align(16, nbpoints * 2 * 3 * sizeof(float));
 
 #ifdef _OPENMP
-#pragma omp parallel default(none) shared(modifier, buf) reduction(min : xm, ym) reduction(max : xM, yM)
+#pragma omp parallel default(none) shared(modifier) reduction(min : xm, ym) reduction(max : xM, yM)
 #endif
     {
 #ifdef _OPENMP
 #pragma omp for schedule(static)
 #endif
       for(int i = 0; i < awidth; i++)
         lf_modifier_apply_subpixel_geometry_distortion(modifier, xoff + i * xstep, yoff, 1, 1, buf + 6 * i);
 ```
Now you please show me where we change `buf` variable :)
If we do change it, then how do we free it afterwards?

https://computing.llnl.gov/tutorials/openMP/#SHARED
```
SHARED Clause

 Purpose:
The SHARED clause declares variables in its list to be shared among all threads in the team.
```
The `SHARED()` is **only** about the variable itself, not about the buffer it points to.
This already works everywhere in darktable.